x/campaign/keeper/msg_create_campaign.go

```diff
@@ -14,6 +14,17 @@ import (
 func (k msgServer) CreateCampaign(goCtx context.Context, msg *types.MsgCreateCampaign) (*types.MsgCreateCampaignResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
+	// check if the metadata length is valid
+	p := k.GetParams(ctx)
+	maxMetadataLength := p.MaxMetadataLength
+	if len(msg.Metadata) > int(maxMetadataLength) {
```

Comment:
```suggestion
	if uint64(len(msg.Metadata)) > maxMetadataLength {
```
we should not use this type of conversation